aws/cloudformation/cloud_formation_stack.yml.erb

```
@@ -410,9 +413,7 @@ Resources:
     Properties:
       Description: 'Lambda for dynamically routing marketing traffic to different origins based on the URL path'
       FunctionName: !Sub "${AWS::StackName}-marketing-router"
-      # The JS file for this function depends on the Sub function to inject the value of MarketingDomainName
-      Code: 
-        ZipFile: !Sub <%=inline_js('lambdas/marketing-router/index.js', uglify: false) %>
+      Code: <%=package_node_lambda('marketing-router', environment_variables: {marketingDomain: INTERNAL_MARKETING_DOMAIN_NAME}) %>
```

Can we use environment variables? Then you can use the cloudformation parameter and avoid the extra ruby variables.

```suggestion
      Code: <%=package_node_lambda('marketing-router') %>
      Environment:
        Variables:
          INTERNAL_MARKETING_DOMAIN_NAME: !Ref InternalMarketingDomain
```

They can then be retrieved like:

```javascript
const marketingDomain  = process.env.INTERNAL_MARKETING_DOMAIN_NAME
```

https://docs.aws.amazon.com/AWSCloudFormation/latest/TemplateReference/aws-resource-lambda-function.html#cfn-lambda-function-environment

---

Unfortunately Lambda@Edge doesn't support environment variables. That was why we were originally using the !Sub hack to get the variable into code.